Expose Page identifier via WKBrowsingContextHandle SPI instead of WKPageRef

pascoej opened the pull request:
#### b791516ec89988ea1a2768a76a803091e0465382
<pre>
Expose Page identifier via WKBrowsingContextHandle SPI instead of WKPageRef
<a href="https://bugs.webkit.org/show_bug.cgi?id=241941">https://bugs.webkit.org/show_bug.cgi?id=241941</a>
rdar://problem/95814743

Reviewed by Chris Dumez.

* Source/WebKit/Shared/API/Cocoa/WKBrowsingContextHandle.mm:
(-[WKBrowsingContextHandle _initWithPageProxyID:andWebPageID:]):
(-[WKBrowsingContextHandle encodeWithCoder:]):
(-[WKBrowsingContextHandle initWithCoder:]):
(-[WKBrowsingContextHandle copyWithZone:]):
(-[WKBrowsingContextHandle description]):
* Source/WebKit/Shared/API/Cocoa/WKBrowsingContextHandleInternal.h:
* Source/WebKit/Shared/API/Cocoa/WKBrowsingContextHandlePrivate.h: Copied from Source/WebKit/Shared/API/Cocoa/WKBrowsingContextHandleInternal.h.
* Source/WebKit/UIProcess/API/C/WKPage.cpp:
(WKPageDispatchActivityStateUpdateForTesting):
(WKPageGetIdentifier): Deleted.
* Source/WebKit/UIProcess/API/C/WKPage.h:
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:
* Source/WebKit/WebProcess/InjectedBundle/API/mac/WKWebProcessPlugInBrowserContextController.mm:
(+[WKWebProcessPlugInBrowserContextController lookUpBrowsingContextFromHandle:]):
* Source/WebKit/WebProcess/cocoa/WebProcessCocoa.mm:
(WebKit::WebProcess::transformHandlesToObjects):

Canonical link: <a href="https://commits.webkit.org/251840@main">https://commits.webkit.org/251840@main</a>
</pre>

cdumez commented on Source/WebKit/Shared/API/Cocoa/WKBrowsingContextHandle.mm:
```diff
@@ -90,12 +90,12 @@ + (BOOL)supportsSecureCoding
 - (id)copyWithZone:(NSZone *)zone
 {
-    return [[WKBrowsingContextHandle allocWithZone:zone] _initWithPageProxyID:_pageProxyID andWebPageID:_webPageID];
+    return [[WKBrowsingContextHandle allocWithZone:zone] _initWithPageProxyID:_pageProxyID andWebPageID:akeObjectIdentifier<WebCore::PageIdentifierType>(_webPageID)];
```
akeObjectIdentifier ?

cdumez commented on Source/WebKit/Shared/API/Cocoa/WKBrowsingContextHandlePrivate.h:
```diff
@@ -0,0 +1,30 @@
+#import <WebKit/WKBrowsingContextHandle.h>
+
+@interface WKBrowsingContextHandle (WKPrivate)
+@property (nonatomic, readonly, getter=_webPageIDUInt64) uint64_t webPageIDUInt64;
```
This doesn't look like a good name. Even though this is SPI, I think the naming in the previous patch (webPageID) was better.

pascoej committed:
Expose Page identifier via WKBrowsingContextHandle SPI instead of WKPageRef
https://bugs.webkit.org/show_bug.cgi?id=241941
rdar://problem/95814743

Reviewed by Chris Dumez.

* Source/WebKit/Shared/API/Cocoa/WKBrowsingContextHandle.mm:
(-[WKBrowsingContextHandle _initWithPageProxyID:andWebPageID:]):
(-[WKBrowsingContextHandle encodeWithCoder:]):
(-[WKBrowsingContextHandle initWithCoder:]):
(-[WKBrowsingContextHandle copyWithZone:]):
(-[WKBrowsingContextHandle description]):
* Source/WebKit/Shared/API/Cocoa/WKBrowsingContextHandleInternal.h:
* Source/WebKit/Shared/API/Cocoa/WKBrowsingContextHandlePrivate.h: Copied from Source/WebKit/Shared/API/Cocoa/WKBrowsingContextHandleInternal.h.
* Source/WebKit/UIProcess/API/C/WKPage.cpp:
(WKPageDispatchActivityStateUpdateForTesting):
(WKPageGetIdentifier): Deleted.
* Source/WebKit/UIProcess/API/C/WKPage.h:
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:
* Source/WebKit/WebProcess/InjectedBundle/API/mac/WKWebProcessPlugInBrowserContextController.mm:
(+[WKWebProcessPlugInBrowserContextController lookUpBrowsingContextFromHandle:]):
* Source/WebKit/WebProcess/cocoa/WebProcessCocoa.mm:
(WebKit::WebProcess::transformHandlesToObjects):

Canonical link: https://commits.webkit.org/251840@main
